src/Microsoft.AspNet.WebUtilities/MultipartReader.cs

```diff
@@ -74,7 +74,7 @@ public MultipartReader(string boundary, Stream stream, int bufferSize)
             return new MultipartSection() { Headers = headers, Body = _currentStream, BaseStreamOffset = baseStreamOffset };
         }
 
-        private async Task<IDictionary<string, StringValues>> ReadHeadersAsync(CancellationToken cancellationToken)
+        private async Task<Dictionary<string, StringValues>> ReadHeadersAsync(CancellationToken cancellationToken)
```

Comment:
It gets added to a HeaderDictionary which uses Dictionary for struct enumerators; if you could enumerate using struct via an interface without boxing the world would be a simpler place :(